app/controllers/users_controller.rb

```diff
@@ -4,7 +4,10 @@ class UsersController < ApplicationController
   before_filter :authenticate_user!, only: [:add_status]
 
   def index
-    @users = User.search(params)
+    count_timezone_degrees
+    @users = User.filter(params.slice(:project_filter, :timezone_filter, :online)).allow_to_display.by_create
```

Comment:
it is generally not a good idea to mutate a data structure, params hash in this case, from within somewhere else, count_timezone_degrees method.
I would prefer to have something like:
`tz_filter = set_time_zone_filter` and then pass it explicitly to User.filter

another concern is passing user submitted params directly to scopes.  I wonder if there is a security problem here.